shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-provider/shardingsphere-transaction-xa-narayana/src/main/java/org/apache/shardingsphere/transaction/xa/narayana/config/NarayanaConfigurationFileGenerator.java

```diff
@@ -98,6 +100,17 @@ private NarayanaConfigEntry createEntry(final String key, final Collection<Strin
         return result;
     }
     
+    private String getRecoveryId(final Properties transactionProps, final InstanceContext instanceContext) {
+        if (transactionProps.containsKey("xaRecoveryNodes")) {
+            Collection<?> xaRecoveryNodes = (Collection<?>) transactionProps.get("xaRecoveryNodes");
+            return Joiner.on(",").join(xaRecoveryNodes);
+        } else {
+            return instanceContext.getInstance().getXaRecoveryIds().isEmpty()
+                    ? instanceContext.getInstance().getInstanceDefinition().getInstanceId()
+                    : Joiner.on(",").join(instanceContext.getInstance().getXaRecoveryIds());
+        }
+    }
+    
```

Comment:
`xaRecoveryNodes` and `nodeIdentifier` may not be configurable. These two configurations are unique to the proxy instance. In cluster mode, the configuration here is shared by all proxies. You can modify `xaRecoveryNodes` through `distsql`.